Lots of update to standardize events page

akibrhast opened the pull request:
This pull request is in relation to #1041 and it

closes #1109 
closes #1114 
closes #1112 
closes #1113 
closes #1111 
closes #1116 

😄 😄

akibrhast committed:
Moved events.html to an events folder

akibrhast committed:
add permalinks and change title

akibrhast committed:
Add style and render days of week with liquid

akibrhast committed:
Adjust size/margin of Our Events and Our Locations

akibrhast committed:
Change css for .event-title

akibrhast committed:
change padding in .getting-started-page-card-hack-nights

akibrhast committed:
change css for .title-meetup class

akibrhast committed:
put subheading and paragraphs in same div and align righ of image

akibrhast committed:
Constraint fixes
- 72px margin between between each Meetup category
- 16px margin between subheadings and paragraphs
- 80px margin between last paragraph and Meetup button

akibrhast committed:
componentize left col

akibrhast committed:
Change heading spacing
- Add 72px margin between heading and first paragraph (same as in the Meetup column card)

akibrhast committed:
Fix Onboarding Link
- Onboarding link leads to wrong url- should be to Getting Started page (https://www.hackforla.org/getting-started)

akibrhast committed:
Make time zone text bold
- Make time zone text bold, should be font-weight: 700
- In .userTimeZone class

akibrhast committed:
Change css for .day-header-1 class
- Make all of the days of week ALL CAPS (MONDAY, TUESDAY, etc. - in the html), bold, and 18px - should be-
    font-size: 1.125rem (18px)
    font-weight: 700
    + in HTML: all-caps titles
- In .day-header-1 class
- Add 30px margin above the days of week and 10px margin belo
    margin: 30px 0 10px 0

akibrhast committed:
Remove padding from .day-header-1 class
- Remove the padding that pushes the meeting times to the right- Add:
	padding: 0
- Add to .day-header-1 class

akibrhast committed:
css changes and refactors
- created new yml files to loop over events in left-col-container.html
- Adjust banner font size, weight, and margin - should be -
	font-size: 1rem         (16px)
	font-weight: 700
	margin: auto 29%

akibrhast committed:
Add mobile view css overrides
- Decrease the padding of the page cards in their collapsed form- should be-
	padding: 5px 15px
- Adjust the border radius- should be -
	border-radius: 20px

akibrhast committed:
merge teddys fix

akibrhast committed:
Some css changes

akibrhast commented:
Hey @jbubar @drubgrubby . Please take a moment to review this pull request
> the page looks good but I have some important edits and haven’t had time to list them out. Also a dev should be added as a reviewer as well, because splitting the page into separate html files is not that common (except for the home page, which was made a long time ago) so I just want to make sure that’s all fine (and they can verify code-related elements that I’m not knowledgable about). But first I just need to get my edits down, and I’m slammed today. I should be able to soon though. - @daniellex0 

Please await @daniellex0 styling edits, 
> so that you can focus on the code!

drubgrubby reviewed (commented):
@akibrhast - Really impressive work.  You really took a lot of different issues on the events page, saw a different way to look at the issue, and then took it apart and put it back together in a new way that solves a lot of the issues.  

I am curious that you deconstructed the page and created all of these files to include, but left the javaScript on the .html page rather than moving it to a .js file and including it. Is there a reason that you left it here? 

That said, it's a different approach to structuring the site and building the pages, and even though I like it personally, I think it needs a discussion about if we want to go in this direction.  

@cnk @ExperimentsInHonesty - When you have a sec, can you weigh in on Akib's restructuring of the way the events page is built? It seems to work, but it's also a change from how we have been doing things, so a discussion seems warranted. 

I'm not looking at any of the design questions.  I will leave that for Danielle.

akibrhast committed:
Moved Javascript
- Moved content in the previously second script tag into righ-col-content.html. Because everything the code within that script
  tag is doing is very tighty tied to everything within the right-col-content.html, i.e the scheduler.

akibrhast committed:
path to refactor

akibrhast committed:
Merge branch 'upstream-gh-pages' into events-multi-col

daniellex0 reviewed (changes requested):
Ok @akibrhast sorry about the delay- thank you for making these changes! Some styling edits before this can go live (I think some of these issues are from a previous update to the page, but I appreciate if you can fix these to make the page look good for now before the other issues are done!):

1. Can you please actually remove the `content-section` class? It's causing the yellow banner to have 16px margin on both sides, which makes it look especially weird in mobile-

<img width="319" alt="Screen Shot 2021-02-25 at 8 58 50 PM" src="https://user-images.githubusercontent.com/67400366/109257117-4aec1c00-77ac-11eb-8b29-9d2aa9c24723.png">

I'll make this into a separate issue so that someone can figure it out after, because there seem to be some potential [solutions](https://css-tricks.com/full-width-containers-limited-width-parents/) (or we might just have to forego this standard class on this page.. not sure).

2. Please align the images with the text in both left and right column cards- 

So instead of this-

<img width="453" alt="Screen Shot 2021-02-25 at 8 44 54 PM" src="https://user-images.githubusercontent.com/67400366/109261770-36605180-77b5-11eb-8c61-a89f6d0b9e5b.png">

it should look closer to this-

<img width="453" alt="Screen Shot 2021-02-25 at 10 04 18 PM" src="https://user-images.githubusercontent.com/67400366/109261891-758ea280-77b5-11eb-809f-127722d7545a.png">

I believe this can be done by putting the text and image in the same div, and their new class should have the following properties (**only for above mobile size**):

```
display: flex;
align-items: center;
```

3. In `.notification-banner p` class please remove the `line-height` property (it should go to default line height)

4. Please make these highlighted changes to the collapsed column cards:

<img width="326" alt="Screen Shot 2021-02-25 at 10 22 28 PM" src="https://user-images.githubusercontent.com/67400366/109263519-3f9eed80-77b8-11eb-83b2-9116438fb65c.png">

5. Please remove these lines:

<img width="287" alt="Screen Shot 2021-02-25 at 10 31 04 PM" src="https://user-images.githubusercontent.com/67400366/109264267-90fbac80-77b9-11eb-8364-461708bb0a4e.png">

6. And this is my bad, I made this edit to the text on Figma but I'm realizing it actually looks too big, so please change the font size here back to 1rem (but keep the html text in all caps) and remove the line height, as seen here:

<img width="262" alt="Screen Shot 2021-02-25 at 10 39 02 PM" src="https://user-images.githubusercontent.com/67400366/109264886-8392f200-77ba-11eb-85b3-e3e54faf00db.png">


That should be good for now, I'll make other edits on the other pages. Please just make sure everything (including my edits here) look fine in mobile before it goes live :)

akibrhast committed:
Major refactor of existing js. This can now be separated into its own .js file

akibrhast committed:
Please align the images with the text in both left and right column cards

akibrhast committed:
In .notification-banner p class please remove the line-height property (it should go to default line height)

akibrhast committed:
Addresses point 4 made in the change request

akibrhast committed:
Addresses point 5 in the change request

akibrhast committed:
Adresses point 6

akibrhast committed:
Adress point 1 made in change request
- Can you please actually remove the content-section class?
  It's causing the yellow banner to have 16px margin on both sides, which makes it look especially weird in mobile-

akibrhast commented:
@daniellex0 Made all the changes as request!

![image](https://user-images.githubusercontent.com/32349001/109269801-0f1c7b00-77db-11eb-812c-11650913f7e9.png)

![image](https://user-images.githubusercontent.com/32349001/109269833-1b083d00-77db-11eb-9e45-b954e06fa86b.png)

daniellex0 reviewed (changes requested):
Thank you for making all of the changes @akibrhast ! Ok, truly, last changes *I promise* (any others will go in the other issues). Sorry, there are just some margin issues that I can't let go live on the site!

Just make these changes to these 4 classes please and you'll be done:

<img width="295" alt="Screen Shot 2021-02-26 at 12 29 34 AM" src="https://user-images.githubusercontent.com/67400366/109276407-d2e11e80-77ca-11eb-8da8-5024beaf141e.png">

margin-bottom instead of margin-top
<img width="199" alt="Screen Shot 2021-02-26 at 12 22 00 AM" src="https://user-images.githubusercontent.com/67400366/109276423-d8d6ff80-77ca-11eb-8157-f8570ed974cf.png">

<img width="238" alt="Screen Shot 2021-02-26 at 12 21 30 AM" src="https://user-images.githubusercontent.com/67400366/109276433-dd031d00-77ca-11eb-9049-47b22a1d7893.png">

Thank you!!

akibrhast commented:
@daniellex0 best ever front end issue i had to deal with.... you tell me exactly what and where needs to be changed ... I go in an change it! 👯‍♂️  👯‍♂️
_Thankfully you are nice enough to take of playing with the css in dev tools and figuring out what works!!!_

akibrhast committed:
I was promised these are the last changes

akibrhast committed:
Multi column events display now working

jbubar commented on events/our-locations-content.html:
```diff
@@ -0,0 +1,177 @@
+        <img class="location-image" src="../assets/images/hack-nights/westside1.png" alt="westside" />
+      </div>
+      <div class="location-content">
+        <h1 style="font-size: 35px; font-weight: normal">Monday</h1>
```
I dont think this was your doing. but there is inline styling here, that prevents the proper styling from showing up.

jbubar replied:
Could you delete these inline styles and style it in ths scss?
something like 
```
.location-content h1{
    font-size: 1.75rem;
    font-weight: 700;
}
```
got that from the [figma](https://www.figma.com/file/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=6777%3A16)

daniellex0 commented:
Styling approved 👍

jbubar commented on events/our-locations-content.html:
```diff
@@ -0,0 +1,177 @@
+        <img class="location-image" src="../assets/images/hack-nights/DTLA.png" alt="DTLA" />
+      </div>
+      <div class="location-content">
+        <h1 style="font-size: 35px; font-weight: normal">Tuesday</h1>
```
also here

jbubar commented on events/our-locations-content.html:
```diff
@@ -0,0 +1,177 @@
+        <img class="location-image" src="../assets/images/hack-nights/southla.png" alt="southLA" />
+      </div>
+      <div class="location-content">
+        <h1 style="font-size: 35px; font-weight: normal">Wednesday</h1>
```
and here

jbubar reviewed (changes requested):
Really really cool stuff. I always learn from reading your code..

A few things that I noticed that were not a part of this issue...
maybe we should have these be scrollable?
![Screen Shot 2021-02-26 at 9 54 11 AM](https://user-images.githubusercontent.com/53061723/109337400-97ba0c00-7819-11eb-95ff-9247ec6cbfbb.png)

and  this does not look so good in mobile view
![Screen Shot 2021-02-26 at 9 57 43 AM](https://user-images.githubusercontent.com/53061723/109337438-a9031880-7819-11eb-86d4-6021fcb74da9.png)

But I know that teddy already created an issue for that.

I will finish my review later.. @akibrhast just informed me that our live wins page is down

akibrhast committed:
Add time zone text

akibrhast commented:
@jbubar I have already created a separate issue that address the change you requested . There is already a LOOOOT of changes on this one pull request(Not proud of it and I am sorry to whoever is having to review this). Should we add more into it?
https://github.com/hackforla/website/issues/1115

---

The our-locations-content.html is a repeatable pattern that needs to be in a for loop somehow. That is why I did not address any of the changes requested in figma for that particular section. I have spoken about this to @daniellex0  and she has agreed on separating it into a separate issue. The same goes for the crappy mobile display, which @TeddyTelanoff made an issue for.
- #1115 
- #1110
- #1106

I would be very averse to doing this

![image](https://user-images.githubusercontent.com/32349001/109340557-2be69c00-7837-11eb-8398-9654160b4dc2.png)

I fear that if I make the change now, the bigger problem of having three hard coded divs, which themselves has 6 to 8 hard coded divs will be kind of forgotten!

thoughts? opinions?

jbubar reviewed (approved):
Thanks Akib for clarifying! and thanks for creating all of those issues!
Everything looks great! I like the new structure!

akibrhast commented:
<details>
<summary>This is what the page looks like after all the changes applied in desktop</summary>

![image](https://user-images.githubusercontent.com/32349001/109396672-b8a95c80-7900-11eb-8a6c-743b3b5a3fdd.png)

</details>
<details>
<summary>This is what the page looks like after all the changes applied in pixel2XL</summary>

![image](https://user-images.githubusercontent.com/32349001/109396711-e5f60a80-7900-11eb-8e4f-51e4b8a18a8a.png)

</details>

jbubar commented:
Here is a similar restructure with the wins page #1132

akibrhast commented:
> @ExperimentsInHonesty - For now we are going to leave  it so such that a new page follows the directory structure as in this pull request.
 


A new issue should be created before merging this, that should be put into icebox and made into a discussion regarding the directory structure going forward at hack4la

jbubar commented:
#1138
